content/docs/how-we-work/baseline/git-workflow.md

```diff
@@ -30,3 +30,25 @@ The branches can be deleted immediately and [automatically](https://docs.github.
 [GitHub provides a variety of protection mechanism](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches) for different branches. It especially recommends that the main branch (often called `main` or `master`) is protected from accidental pushes.
 
 We recommend that the default branch is always protected. At a minimum, a PR should be required before merging to the main branch. You can also enforce conditions such as the PR must have approvals and requiring reviews from code owners. More rules can be found in the [documentation](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches).
+
+### Number of branches
+
+It is recommended to have minimal number of branches as highlighted above. However below can be some scenarios based on which you can decide what works for your project.
+
+#### Scenario 1 (1 branch)
+
+This is the most ideal scenario where we have only 1 branch which is either `main` or `master`. Any other branches are temporary feature branches which would be deleted after the PR is closed. In case of any feature that needs to be reverted, raising a revert PR against the merged one would work.The `main` branch is linked to the Development environment where the next code for Production is pushed and QA is done followed by bug fixes.
+
+This method is effective for projects with few or no feature changes post User Acceptance Testing (UAT).However, we should withhold merging any tickets into the `main` or `master` branch if they are not planned current release. Instead, these tickets should remain on a feature branch until the appropriate tag is created for the planned deployment continuously rebased against `main` or `master`.
```

Comment:
@hussainweb Let me explain the scenario where this is happening currently.
1. We have a set of features grouped that are planned in the current release.
2. There is more capacity available in the team and they pick up items from the product backlog that are not on hold but are currently being worked on. These are WIP developments and hence in future releases.

Point 2 is done when we have more Continuous development (CD) hours in large sizes like in MSF so that we work on multiple releases at the same time to utilise the CD hours. Above is the use case, when the work is not on hold, currently being developed.

I agree that ideally, we need to have a feature flag so that this scenario does not happen and we continuously merge things to the `main`. We plan to work on a couple of projects to start with feature flags and gradually roll it out.

I have removed the line in 63e7758aba7240ce4a932805380436a7af4de5c2 as anyway we have mentioned about features staying in feature environments in your suggestion above.